zwave_js_server/protocol/controller.py

```diff
@@ -0,0 +1,90 @@
+"""Provide a protocol for the controller model."""
+from enum import Enum
+from typing import TYPE_CHECKING
+
+from ..model.node import Node
+
+if TYPE_CHECKING:
+    from ..model.controller import Controller, ControllerEvent
+
+
+class Type(Enum):
+    """Represent a controller event type."""
+
+    INCLUSION_FAILED = "inclusion failed"
+    EXCLUSION_FAILED = "exclusion failed"
+    INCLUSION_STARTED = "inclusion started"
+    EXCLUSION_STARTED = "exclusion started"
+    INCLUSION_STOPPED = "inclusion stopped"
+    EXCLUSION_STOPPED = "exclusion stopped"
+    NODE_ADDED = "node added"
+    NODE_REMOVED = "node removed"
+    HEAL_NETWORK_PROGRESS = "heal network progress"
+    HEAL_NETWORK_DONE = "heal network done"
+
+
+class Handler:
```

Comment:
I see what's going on here, but it is this really improving the existing logic? I think it's harder to follow. 

Instead of having a Controller instance that has controller properties and methods, we know move the methods to a second class and have that class mutate the public instance variables of a controller? That kinda treats a controller like a C-like struct: just a holder of data. 

These methods are changing instance variables of the models and `event.data`. It fels weird to decouple that. A protocol mutates data and the owner class is not in the loop?

Reply:
I think that we can decide to have the `get_handler` logic be included in the models that receive events so that we can still split out handling of each event into different methods.

Reply:
I think that works.

Reply:
I've moved the handlers to each model class.